src/index.ts

```diff
@@ -1,162 +1,625 @@
+    session: () => Promise<Maybe<Session>>
+  }
+>
+
+
+export const strategyTypes = {
+  default: BaseAuth.TYPE,
+  webCrypto: BaseAuth.TYPE
+}
 
-import { AppInitOptions, InitialisationError, PermissionedAppInitOptions } from "./init/types.js"
-import { validateSecrets } from "./auth/state.js"
-import { bootstrapFileSystem, loadFileSystem } from "./filesystem.js"
 
-import FileSystem from "./fs/index.js"
 
-import { impl as auth } from "./auth/implementation.js"
-import { isPermissionedAppState } from "./auth/state.js"
-import { setImplementations } from "./setup.js"
-import { BASE_IMPLEMENTATION } from "./auth/implementation/base.js"
-import { USE_WNFS_IMPLEMENTATION } from "./auth/implementation/use-wnfs.js"
-import { LOBBY_IMPLEMENTATION } from "./auth/implementation/lobby.js"
-import { AppState } from "./auth/state/app.js"
-import { PermissionedAppState } from "./auth/state/permissionedApp.js"
-import * as appState from "./auth/state/app.js"
-import * as permissionedAppState from "./auth/state/permissionedApp.js"
+// ENTRY POINTS
 
 
 /**
- * Check if we're authenticated and initiate the user's file system if 
- * authenticated (can be disabled).
+ * Build a webnative program.
  *
- * See `loadFileSystem` if you want to load the user's file system yourself.
- * NOTE: Only works on the main/ui thread, as it uses `window.location`.
+ * This will give you a `Program` object which has the following properties:
+ * - `session`, a `Session` object if a session was created before.
+ * - `auth`, a means to control the various auth strategies you configured. Use this to create sessions. Read more about auth components in the toplevel `auth` object documention.
+ * - `confidences`, a means to control confidences. Use this to collect & request confidences, and to create a session based on them. Read more about confidences in the toplevel `confidences` object documentation.
+ * - `components`, your full set of `Components`.
  *
+ * See `assemble` for more information.
  */
-export async function app(options: AppInitOptions): Promise<AppState> {
-  options = options || {}
+export async function program(settings: Partial<Components> & Configuration): Promise<Program> {
+  if (!settings) throw new Error("Expected a settings object of the type `Partial<Components> & Configuration` as the first parameter")
 
-  const { useWnfs = false } = options
+  const components = await gatherComponents(settings)
+  return assemble(settings, components)
+}
 
-  if (useWnfs) {
-    setImplementations(USE_WNFS_IMPLEMENTATION)
-  } else {
-    setImplementations(BASE_IMPLEMENTATION)
-  }
 
-  // Check if browser is supported
-  if (globalThis.isSecureContext === false) throw InitialisationError.InsecureContext
-  if (await isSupported() === false) throw InitialisationError.UnsupportedBrowser
 
-  const authedUsername = await common.authenticatedUsername()
+// PREDEFINED COMPONENT COMBINATIONS
```

Comment:
See this section for a detailed explanation on the components and how they work together. A short explanation is located in the changelog.